docs/architecture/cloud-native/azure-security.md

```diff
@@ -0,0 +1,257 @@
+
+## Role-based access control for restricting access to Azure resources
+
+RBAC is a system that provides an identity to applications running in Azure. Applications can access resources using this identity instead of or in addition to using keys or passwords. 
+
+## Security Principals
+
+The first component in RBAC is a security principal. A security principal can be a user, group, service principal, or managed identity. 
+
+![Figure 10-1 Different types of security principals](media/rbac-security-principal.png)
+
+* User - Any user who has an account in Azure Active Directory is a user.
+
+* Group - A collection of users from Azure Active Directory. As a member of a group, a user takes on the roles of that group in addition to their own.
+
+* Service principal - A security identity under which services or applications run.
+
+* Managed identity - An Azure Active Directory identity managed by Azure. Managed identities are typically used when developing cloud applications that manage the credentials for authenticating to Azure services.
+
+The security principal can be applied to most any resource. This means that it's possible to assign a security principal to a container running within Azure Kubernetes, allowing it to access secrets stored in Key Vault. An Azure Function could take on a permission allowing it to talk to an Active Directory instance to validate a JWT for a calling user. Once services are enabled with a service principal, their permissions can be managed granularly using roles and scopes.  
+
+## Roles
+
+A security principal can take on many roles or, using a more sartorial analogy, wear many hats. Each role defines a series of permissions such as "Read messages from Azure Service Bus endpoint". The effective permissions of a security principal is the combination of all the permissions assigned to all the roles that security principal has. Azure has a large number of built-in roles and users can define their own roles.
```

Comment:
The effective permissions of a security principal is -> should it be permission?

Reply:
I'll make it 'permission set'